src/main/groovy/com/bmuschko/gradle/docker/tasks/container/DockerWaitContainer.groovy

```diff
@@ -15,19 +15,37 @@
  */
 package com.bmuschko.gradle.docker.tasks.container
 
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.Optional
+
+import java.util.concurrent.TimeUnit
+
 class DockerWaitContainer extends DockerExistingContainer {
 
     protected int exitCode
 
-    DockerWaitContainer(){
-        ext.getExitCode = {exitCode}
+    /**
+     * Wait timeout in seconds.
+     */
+    @Input
+    @Optional
+    Long timeout
```

Comment:
For future reference: we would normally not want to add in additional features on top of the migration to docker-java. Try to keep PR's as terse as possible and each additional change addressed in a separate PR.

With that said, I don't mind sneaking this in now just keep this in mind for future work.